exercises/template/ex02_oo_basics/src/shape.cpp

```diff
@@ -0,0 +1,76 @@
+#include "shape.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+ostream &operator<<(ostream &os, const unique_ptr<Shape> &shape) {
```

Comment:
You could implement streaming for each shape so we could see the radius for circles or base and height for triangles for example. However I am not too bothered.